tobler/tests/test_diagnostics.py

```diff
@@ -0,0 +1,17 @@
+"""test diagnostics funtions"""
+
+import geopandas
+from tobler.diagnostics import _smaup
+from libpysal.weights import Queen
+from libpysal.examples import load_example
+
+
+sac1 = load_example("Sacramento1")
+sac1 = geopandas.read_file(sac1.get_path("sacramentot2.shp"))
+sac1["pct_poverty"] = sac1.POV_POP / sac1.POV_TOT
+
+
+def test_smaup():
+    queen = Queen.from_dataframe(sac1)
+    s = _smaup(1,sac1["pct_poverty"].to_numpy(),queen)
```

Comment:
Why is the first argument 1 here?

I'm wondering what the test is checking?

Reply:
The first argument here is the number of regions, `k`. It's 1 to show `smaup` that the geometry should only be aggregated into one region. `smaup` uses its own logic on how to [aggregate the regions contiguously](https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0207377#sec003) on different values of `k`.

The test is checking that we get a valid result from `smaup` so that we can check the `s.summary` in `smaup_kwds`. A more complete test would probably use `smaup` in a more accurate context of checking if the distributional characteristics hold over a larger spatial aggregation - like the 7 counties that make up the Sacramento MSA.

Reply:
In the most recent commits, I've rewritten `smaup` a little bit so that it always takes k from the length of `target_df`, and so that this test now runs on the tracts -> precincts interpolation from the notebook examples.